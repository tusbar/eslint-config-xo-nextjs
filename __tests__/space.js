const {runEslint} = require('./__helpers__/eslint')

describe('index', () => {
  it('should generate spacing errors messages', () => {
    const statements = [
      'const el = (\n<div>\n<span />\n</div>\n)',
      'const el = (\n\t<div>\n\t\t<span />\n\t</div>\n)'
    ]

    for (const statement of statements) {
      const {messages} = runEslint(statement)

      expect(messages).toMatchSnapshot()
    }
  })
})
