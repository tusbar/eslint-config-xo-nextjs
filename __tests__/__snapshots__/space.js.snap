// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index should generate spacing errors messages 1`] = `
Array [
  Object {
    "column": 7,
    "endColumn": 9,
    "endLine": 1,
    "fix": Object {
      "range": Array [
        6,
        8,
      ],
      "text": "element",
    },
    "line": 1,
    "message": "The variable \`el\` should be named \`element\`. A more descriptive name will do too.",
    "nodeType": "Identifier",
    "ruleId": "unicorn/prevent-abbreviations",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 6,
    "endLine": 2,
    "line": 2,
    "message": "'React' must be in scope when using JSX",
    "nodeType": "JSXOpeningElement",
    "ruleId": "react/react-in-jsx-scope",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 6,
    "endLine": 2,
    "fix": Object {
      "range": Array [
        13,
        13,
      ],
      "text": "	",
    },
    "line": 2,
    "message": "Expected indentation of 1 tab character but found 0.",
    "nodeType": "JSXOpeningElement",
    "ruleId": "react/jsx-indent",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 9,
    "endLine": 3,
    "line": 3,
    "message": "'React' must be in scope when using JSX",
    "nodeType": "JSXOpeningElement",
    "ruleId": "react/react-in-jsx-scope",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 9,
    "endLine": 3,
    "fix": Object {
      "range": Array [
        19,
        19,
      ],
      "text": "	",
    },
    "line": 3,
    "message": "Expected indentation of 1 tab character but found 0.",
    "nodeType": "JSXOpeningElement",
    "ruleId": "react/jsx-indent",
    "severity": 2,
  },
]
`;

exports[`index should generate spacing errors messages 2`] = `
Array [
  Object {
    "column": 7,
    "endColumn": 9,
    "endLine": 1,
    "fix": Object {
      "range": Array [
        6,
        8,
      ],
      "text": "element",
    },
    "line": 1,
    "message": "The variable \`el\` should be named \`element\`. A more descriptive name will do too.",
    "nodeType": "Identifier",
    "ruleId": "unicorn/prevent-abbreviations",
    "severity": 2,
  },
  Object {
    "column": 2,
    "endColumn": 7,
    "endLine": 2,
    "line": 2,
    "message": "'React' must be in scope when using JSX",
    "nodeType": "JSXOpeningElement",
    "ruleId": "react/react-in-jsx-scope",
    "severity": 2,
  },
  Object {
    "column": 3,
    "endColumn": 11,
    "endLine": 3,
    "line": 3,
    "message": "'React' must be in scope when using JSX",
    "nodeType": "JSXOpeningElement",
    "ruleId": "react/react-in-jsx-scope",
    "severity": 2,
  },
]
`;
